rfc/rfc-0007-resource-references.md

```diff
@@ -0,0 +1,40 @@
+# RFC-0007: Resource References
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1370](https://github.com/projectriff/riff/pull/1370)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** [RFC 0006: Guidance for child resource names](https://github.com/projectriff/riff/pull/1369)
+
+
+## Problem
+It's common for one resource to reference another. For example, a Deployer can reference a Function build to obtain the latest image, or a stream provider references the gateway Service it created. riff should have a common way to express relationships between Kubernetes resources.
+
+### Anti-Goals
+This RFC is not define how specific riff resources reference other resources, it is only providing guidance.
+
+## Solution
+The Kubernetes API project provides two mechanisms for referencing other resources. ObjectReference and LocalObjectReference.
```

Comment:
You mention "ad hoc" references later, and i assume that we use "ad hoc" references in some form today - shouldn't we include that as one of the ways to create references, and provide an example of that and an explanation of what the issues are with that?

Reply:
by an ad hoc reference, I mean a resource reference that don't follow the proposed solution for referencing objects. Since that's not part of the proposed solution, it doesn't make sense to me to include it under the solution. The aim of this RFC is to minimize the number of references not-following the proposed solution compared to the number that do over time.